includes/settings/class-wcpdf-settings-debug.php

```diff
@@ -904,6 +906,144 @@ private function get_settings_sections(): array {
+			
+			update_option( $option_name, $results );
+			
+			if ( 0 !== $total_limit && $total_fetched >= $total_limit ) {
+				as_unschedule_all_actions( $hook );
+				update_option( $option_name . '::last_time', time() );
+				break; // exit the loop if total limit is reached
+			}
+			
+			$offset += $chunk_size; // increase the offset for the next chunk
+			
+			$args = array(
+				'table_name'    => $table_name,
+				'orderby'       => $orderby,
+				'order'         => $order,
+				'limit'         => $limit,
+				'chunk_size'    => $chunk_size,
+				'offset'        => $offset,
+				'total_fetched' => $total_fetched,
+			);
+			
+			as_enqueue_async_action( $hook, $args );
+		}
+	}
+	
+	public function ajax_numbers_data() {
+		check_ajax_referer( 'wpo_wcpdf_debug_nonce', 'nonce' );
+		
+		$request = stripslashes_deep( $_POST );
+		
+		if ( isset( $request['action'] ) && 'wpo_wcpdf_numbers_data' === $request['action'] && isset( $request['operation'] ) && isset( $request['table_name'] ) ) {
+			extract( $this->filter_fetch_request_data( $request ) );
+			
+			$this->delete_number_table_data( $table_name ); // both operations require delete
+			
+			if ( 'fetch' === $request['operation'] ) {
+				$this->fetch_number_table_data( $table_name, $orderby, $order, $limit );
+			}
+
+			wp_send_json_success( array( esc_url_raw( admin_url( 'admin.php?page=wpo_wcpdf_options_page&tab=debug&section=numbers&orderby=' . $orderby . '&order=' . $order . '&table_name=' . $table_name ) ) ) );
+		} else {
+			wp_send_json_error( array( __( 'Invalid request', 'woocommerce-pdf-invoices-packing-slips' ) ) );
+		}
+	}
+	
+	public function filter_fetch_request_data( $request_data ) {
+		return array(
+			'table_name' => isset( $request_data['table_name'] ) && in_array( $request_data['table_name'], array_keys( $this->get_number_store_tables() ) ) ? sanitize_text_field( $request_data['table_name'] )            : null,
+			'order'      => isset( $request_data['order'] )      && in_array( strtolower( $request_data['order'] ), array( 'desc', 'asc' ) )                ? sanitize_text_field( strtolower( $request_data['order'] ) )   : 'desc',
+			'orderby'    => isset( $request_data['orderby'] )    && in_array( strtolower( $request_data['orderby'] ), array( 'id' ) )                       ? sanitize_text_field( strtolower( $request_data['orderby'] ) ) : 'id',
+			'limit'      => isset( $request_data['limit'] )      && is_numeric( $request_data['limit'] ) && $request_data['limit'] > 0                      ? absint( $request_data['limit'] )                              : 'all',
+		);
+	}
+	
+	public function delete_number_table_data( $table_name ) {
+		if ( empty( $table_name ) ) {
+			return;
+		}
+		
+		delete_option( "wpo_wcpdf_number_data::{$table_name}" );
+		delete_option( "wpo_wcpdf_number_data::{$table_name}::last_time" );
+	}
+	
+	public function search_number_in_table_data( $table_name, $search ) {
```

Comment:
I think it would be better if the search happens directly in the database instead of fetched data.
For a customer with hundreds of orders, it might need to fetch all orders and then perform the search, and also, if they fetch some orders and search and get no result, it could be confusing for them as they think they don't have the searched number.

Reply:
Great idea!